tests/integration-tests/tests/storage/test_deletion_policy.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+
+import boto3
+import pytest
+from assertpy import assert_that
+from botocore.exceptions import ClientError
+from utils import get_compute_nodes_instance_ids, get_root_volume_id
+
+from tests.tags.test_tag_propagation import get_head_node_instance_id
+
+
+@pytest.mark.usefixtures("instance", "scheduler")
+def test_retain_on_deletion(pcluster_config_reader, clusters_factory, region, os):
```

Comment:
can we extract resource cleanup in a fixture in order to make it more robust and attempt a cleanup also when assertions fail? An idea could be to have the following fixture

```
@pytest.fixture
def cleanup_retained_resources_for_cluster():
    cluster = None
    def _register_cluster(c):
        cluster = c

   yield _register_cluster

   # do the cleanup by searching for resources and deleting them (if one of them fails to be deleted let's not stop deletion of the others)
```

Reply:
Done by doing the clean up in cluster_factory